shell/platform/windows/flutter_windows_view.cc

```diff
@@ -50,10 +52,29 @@ void FlutterWindowsView::SetEngine(
                     binding_handler_->GetDpiScale());
 }
 
-void FlutterWindowsView::OnWindowSizeChanged(size_t width,
-                                             size_t height) const {
-  surface_manager_->ResizeSurface(GetRenderTarget(), width, height);
+void FlutterWindowsView::OnWindowSizeChanged(size_t width, size_t height) {
+  std::unique_lock<std::mutex> lock(resize_mutex_);
+  pending_width_ = width;
+  pending_height_ = height;
   SendWindowMetrics(width, height, binding_handler_->GetDpiScale());
+  resize_waiting_ = true;
+
+  if (width > 0 && height > 0) {
+    // for non empty frame block platform thread until GetFrameBufferId is
+    // called with requested size followed by SwapBuffers()
+    resize_cond_.wait(lock, [&] { return !resize_waiting_; });
+  }
+}
+
+uint32_t FlutterWindowsView::GetFrameBufferId(size_t width, size_t height) {
+  std::unique_lock<std::mutex> lock(resize_mutex_);
+
+  if (pending_width_ == width && pending_height_ == height) {
```

Comment:
Can you wrap this block with `if (resize_waiting_) {}`? That way its clear that `pending_width_` and `pending_height_` are only valid during resize.